packages/trader/src/AppV2/Components/TradeParameters/Duration/duration.tsx

```diff
@@ -1,28 +1,64 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import clsx from 'clsx';
 import { observer } from 'mobx-react';
-import { TextField } from '@deriv-com/quill-ui';
+import { ActionSheet, TextField } from '@deriv-com/quill-ui';
 import { getUnitMap } from '@deriv/shared';
-import { Localize } from '@deriv/translations';
+import { Localize, localize } from '@deriv/translations';
 import { useTraderStore } from 'Stores/useTraderStores';
+import DurationActionSheetContainer from './container';
 
 type TDurationProps = {
     is_minimized?: boolean;
 };
 
 const Duration = observer(({ is_minimized }: TDurationProps) => {
-    const { duration, duration_unit } = useTraderStore();
+    const { duration, duration_unit, expiry_time, expiry_type } = useTraderStore();
     const { name_plural, name } = getUnitMap()[duration_unit] ?? {};
     const duration_unit_text = name_plural ?? name;
+    const [selected_hour, setSelectedHour] = useState<number[]>([]);
+    const [is_open, setOpen] = useState(false);
+
+    useEffect(() => {
+        if (duration_unit === 'm' && duration > 59) {
+            const hour = Math.floor(duration / 60);
+            const minutes = duration % 60;
+            setSelectedHour([hour, minutes]);
+        }
+    }, [duration, duration_unit]);
+
+    const getInputValues = () => {
+        if (expiry_type == 'duration') {
+            if (selected_hour.length > 0) {
+                return `${selected_hour[0]} ${localize('hours')} ${selected_hour[1]} ${localize('minutes')} `;
+            }
+            return `${duration} ${duration_unit_text}`;
+        }
+        return `${localize('Ends at')} ${expiry_time} GMT`;
+    };
```

Comment:
```
    const getInputValues = () => {
        if (expiry_type == 'duration') {
            return selected_hour.length > 0) 
                 ?`${selected_hour[0]} ${localize('hours')} ${selected_hour[1]} ${localize('minutes')} `;
                 :`${duration} ${duration_unit_text}`;
        }
        return `${localize('Ends at')} ${expiry_time} GMT`;
    };
```
    
    Up to you, same thing

Reply:
I don't see any help/advantage for adding this . I am just making use of return more to make it more clearer to understand